src/node/mod.rs

```diff
@@ -13,35 +13,69 @@ use channel::network_channel;
 pub use controller::{Controller, Packet};
 pub use header::Header;
 
+use crate::node::state::State;
 pub use state::{MemoryState, SledDiskState};
+use std::net::SocketAddr;
+use std::str::FromStr;
 use std::sync::Arc;
 use tokio::net::TcpStream;
 use tokio::sync::Mutex;
 use tracing::info;
 pub use wire::Wire;
 
-pub async fn node_with_single_peer(address: &str) -> anyhow::Result<()> {
+pub async fn node_with_autodiscovery(addresses_override: Option<String>) -> anyhow::Result<()> {
     let network = Network::Live;
     // let state = SledDiskState::new(Network::Live);
     let state = MemoryState::new(network);
 
     let state = Arc::new(Mutex::new(state));
-    let address = address.to_owned();
-
-    // TODO: peering.nano.org
-
-    let _state_clone = Arc::clone(&state);
-    info!("Spawning a channel to {}", &address);
-    let handle = tokio::spawn(async move {
-        let stream = TcpStream::connect(&address).await.unwrap();
-        // let mut channel = Channel::new(network, state_clone, stream).await;
-        // channel.run().await.unwrap();
-        network_channel(network, state, stream)
+    let configured_peers = if addresses_override.is_some() {
+        parse_socket_list(addresses_override.unwrap())
+    } else {
+        tokio::net::lookup_host("peering.nano.org:7075")
             .await
-            .expect("Error in network_channel");
-    });
+            .unwrap()
+            .collect::<Vec<SocketAddr>>()
+    };
+    state.lock().await.add_peers(configured_peers).await?;
 
-    handle.await.unwrap();
+    let mut handles = vec![];
+    let initial_peers = state.lock().await.peers().await?;
+    for socket_addr in initial_peers {
+        info!("Spawning a channel to {}", socket_addr);
+        let state = state.clone();
+        let handle = tokio::spawn(async move {
+            let stream = TcpStream::connect(socket_addr).await.unwrap();
+            network_channel(network, state, stream)
+                .await
+                .expect("Error in network_channel")
```

Comment:
The `expect()` here is probably reasonable OTOH. In the future we'll handle this gracefully.